### PR TITLE
HPCC-16887 Control TxSummary level in ESP log

### PR DESCRIPTION
**Pull request by wangkx**
Add txSummaryLevel and txSummaryResourceReq to esp.xml and
use them to control TxSummary level and whether log TxSummary
for ESP resource requests or not.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review. I may not catch all of the front end requests and not sure how.

File: esp/bindings/http/platform/httpservice.cpp [hunk 1]
```diff
@@ -329,11 +329,17 @@ int CEspHttpServer::processRequest()
                     methodName.setCharAt(methodName.length()-1, 0);
                 if (!stricmp(methodName.str(), "files"))
                 {
+                    if (!getTxSummaryFrontEndReq())
```
Comment (afishbeck) [inline esp/bindings/http/platform/httpservice.cpp]: Instead of FrontEndReq, perhaps ResourceReq?

**Review by afishbeck — COMMENTED** [review]
Other than a minor naming suggestion (took me a while to figure out what the current name meant) looks fine.

File: esp/bindings/http/platform/httpservice.cpp (hunk 1 above)
Comment (wangkx) [inline esp/bindings/http/platform/httpservice.cpp]: Good suggestion. I will change it.

**Comment by wangkx** [comment]
@afishbeck I just renamed FrontEndReq to ResourceReq. Please review.

**Review by afishbeck — APPROVED** [review]
Looks good.

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged. Also please merge: https://github.com/hpcc-systems/LN/pull/825.

**Comment by richardkchapman** [comment]
@wangkx Given the name change, the commit message should be updated.

**Comment by wangkx** [comment]
@richardkchapman the commit message has been updated to match the name change.

**Pull request merged**